common/badBrowserCheck.ts

```diff
@@ -0,0 +1,31 @@
+import Modernizr from 'modernizr';
+
+const features: (keyof typeof Modernizr)[] = ['flexbox', 'flexwrap', 'localstorage'];
+
+/**
+ * Checks for browser support in the specified feature. If the feature is not supported, the `BadBrowser` overlay will
+ * be shown.
+ *
+ * @param feature
+ * @return {boolean}
+ */
+const isSupported = (feature: keyof typeof Modernizr): boolean => {
+  if (!Modernizr[feature]) {
+    const element = document.getElementsByClassName('BadBrowser')[0];
+    element.className += ' is-open';
+
+    if (/iPhone|iPad|iPod|Android/i.test(navigator.userAgent)) {
+      element.className += ' is-mobile';
+    }
+
+    console.log(`Feature '${feature}' not supported.`);
+
+    return false;
+  }
+
+  return true;
+};
+
+document.addEventListener('DOMContentLoaded', () => {
```

Comment:
Is it possible to run the check before?
eg. do we really need to wait for React to load.

Reply:
’m suspicious of the hidden dependency.
We add a class if a feature is missing, and somewhere in the code there is a listener of the class that will render a view.
Instead of using a class to make the switch maybe we could be more declarative and append to the DOM right away?
This way we don’t need to wait for the rest of the DOM to be loaded (edite